packages/eslint-plugin/src/stylex-no-unused.js

```diff
@@ -0,0 +1,214 @@
+    node &&
+    node.type === 'CallExpression' &&
+    isStylexCallee(node.callee) &&
+    node.arguments.length === 1 &&
+    node.arguments[0].type === 'ObjectExpression'
+  );
+}
+
+function getPropertiesByName(node: Node | null) {
+  const properties = new Map<any, PropertyValue>();
+  if (node == null) {
+    return properties;
+  }
+  node.properties
+    ?.filter((property) => !property.computed && !property.method)
+    .forEach((property) => {
+      const { key } = property;
+      if (key?.type === 'Identifier') {
+        properties.set(key.name, property);
+      } else if (key?.type === 'Literal') {
+        properties.set(key.value, property);
+      }
+    });
+  return properties;
+}
+
+const stylexNoUnused = {
+  meta: {
+    fixable: 'code',
+  },
+  create(context: Rule.RuleContext): { ... } {
+    const stylexProperties = new Map<string, Map<any, PropertyValue>>();
+
+    function saveStylexCalls(node: Node) {
+      const id = node.id;
+      const init = node.init;
+      if (id && id.type === 'Identifier' && init && isStylexDeclaration(init)) {
+        stylexProperties.set(
+          id.name,
+          getPropertiesByName(
+            init.arguments && init.arguments?.length > 0
+              ? init.arguments[0]
+              : null,
+          ),
+        );
+      }
+    }
+    function checkArguments(
+      namespaces: Map<any, PropertyValue>,
+    ): (argument: Expression | SpreadElement | null) => void {
+      return function (argument: Expression | SpreadElement | null): void {
+        if (argument) {
+          if (argument.type === 'Literal') {
+            namespaces.delete(argument.value);
+          } else if (argument.type === 'ObjectExpression') {
+            argument.properties.forEach((property) => {
+              if (property.key) {
+                namespaces.delete(property.key.name);
+              }
+            });
+          } else if (argument.type === 'ArrayExpression') {
+            argument.elements.forEach((element) => {
+              namespaces.delete(element?.value);
+            });
+          } else if (argument.type === 'ConditionalExpression') {
+            const { consequent, alternate } = argument;
+            // check for nested expressions
+            checkArguments(namespaces)(consequent);
+            checkArguments(namespaces)(alternate);
+          } else if (
+            argument.type === 'LogicalExpression' &&
+            argument.operator === '&&'
+          ) {
+            // check for nested expressions but only on the right
+            checkArguments(namespaces)(argument.right);
+          }
+        }
+      };
+    }
+
+    return {
+      Program(node: Program) {
+        // stylex.create can only be singular variable declarations at the root
+        // of the file so we can look directly on Program and populate our set.
+        node.body
+          .filter(({ type }) => type === 'VariableDeclaration')
+          .map(({ declarations }) =>
```

Comment:
You should also look for any top level consts so you can resolve objects passed into `stylex.create` that uses a local const.

Reply:
I don't think I understand you completely, do you mean something like:
```
const field = {
  backgroundColor: 'white',
};

const styles = stylex.create({
  main: {
    height: 4,
  },
  field,
});
```

Reply:
Yes. Something like that. 

Or even the entire object that's passed into `stylex.create`.

Reply:
Got it, however, when testing it out inside doc folder, I realize both of the format are actually illegal:
<img width="866" alt="Screenshot 2024-11-07 at 12 32 40 PM" src="https://github.com/user-attachments/assets/6f6b57cb-7338-410c-bfa8-bf38d4fc6e8c">
<img width="483" alt="image" src="https://github.com/user-attachments/assets/3cec10f7-4ebc-4d7f-b729-994977c1e911">
<img width="740" alt="image" src="https://github.com/user-attachments/assets/5c4b8bbe-4b7e-4452-b294-879314706463">

Reply:
Yeah, we need to fix that rule too. We can punt on this for now.